src/Microsoft.Extensions.Logging.Filter/Internal/FilterLoggerFactory.cs

```diff
@@ -22,12 +24,14 @@ public void AddProvider(ILoggerProvider provider)
 
         public ILogger CreateLogger(string categoryName)
         {
-            return _innerLoggerFactory.CreateLogger(categoryName);
+            throw new NotImplementedException();
```

Comment:
This is one of the main reasons why I think that loggers and providers should be split. I know my other PR is way too large - and probably even too large to just read through it. Sorry for that.

This whole thing just feels really weird to me now. I also know I started this, but I think we should discuss the general approach first before we continue with this.

*Is it ok, that someone can call .WithFilter() and then not use the return object? (see my example above)*

*How does it work if someone does use the returned FilterLoggerFactory and adds it to DI? It will then get resolved whenever someone wants an ILoggerFactory and it will blow up If we throw a NotImplementedException() in CreateLogger*

It just seems that there are too many ways how this could go wrong :-(